src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -182,6 +186,95 @@ public static List<string> GetHeaderValues(HttpRequestHeaders headers, string he
             return list;
         }
 
+        private static bool IsWithinRetryRange(HttpStatusCode statusCode)
+        {
+            // Retry on http 'server error' codes (5xx) as well as redirect
+
+            var rawStatus = (int)statusCode;
+            if (rawStatus == 302)
+                return true;
+            else if (rawStatus >= 500 && rawStatus <= 599)
```

Comment:
I believe we've also seen errors on the client side with status codes from 400-499. Do we want to retry on these as well?